### DISCUSSION TITLE
Possibility to export recognition model to tensorflowjs?

### DISCUSSION THREAD

**Discussion by lieding** [mindee/doctr#1580] [Q&A]
Hi, I have found your tfjs OCR demo, which  I find very interesting and now I have some business tasks, which need to introduce some OCR tools running in browser, your guys make a great job.

But in the demo, the recognition quality is limited. I seara the code, which shows in the demo, it uses  a rather old recognition mode, which runs fast but in limited precision. I have tried switching recognition model to CRNN_VGG16_BN, and I fave followed the recommended process to export as onnx mode, conversion to TF format. But after using `tensorflowjs_converter`, it always failed because of `ValueError: Signature "serving_default" does not exist in the saved model`. I repeated a few times but always the same error. So I post this issue to confirm, if it is possible to replace existing recognition model, or currently this is absolutely not possible.

**Comment by felixdittrich92** [comment]
Hi @lieding :wave:,

Two points:

1. You can add the signature on your own if it doesn't exist: https://stackoverflow.com/a/71259690
2. The docTR onnx exported `detection` models outputs the `logits` vector so to get the final map you need to pass the output to a `sigmoid` function - for recognition models no additional step is required you can pass the output directly to the postprocessor

Hope this helps :)
